src/Integration/Exclusions/ExclusionSettingsFileStorage.cs

```diff
@@ -17,23 +17,84 @@
  * along with this program; if not, write to the Free Software Foundation,
  * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  */
-
-using System.ComponentModel.Composition;
-using SonarLint.VisualStudio.Core.Analysis;
+using System;
+using System.IO;
+using System.IO.Abstractions;
+using Newtonsoft.Json;
+using SonarLint.VisualStudio.Core.Binding;
+using SonarLint.VisualStudio.Core.Exclusions;
+using SonarLint.VisualStudio.Core.Helpers;
+using SonarLint.VisualStudio.Integration.NewConnectedMode;
+using SonarLint.VisualStudio.Integration.Resources;
 using SonarQube.Client.Models;
 
 namespace SonarLint.VisualStudio.Integration.Exclusions
 {
-    [Export(typeof(IExclusionSettingsFileStorage))]
     internal class ExclusionSettingsFileStorage : IExclusionSettingsFileStorage
     {
-        public void SaveSettings(ServerExclusions settings)
+        private readonly BindingConfiguration bindingConfiguration;
+        private readonly ILogger logger;
+        private readonly IFileSystem fileSystem;
+
+        public ExclusionSettingsFileStorage(ILogger logger, IFileSystem fileSystem, BindingConfiguration bindingConfiguration)
```

Comment:
How is this class going to be instantiated by AnalyzableFileIndicator?
I think that maybe we should split between the read and the write. The read takes the bindingConfiguration only to write logging:
1) is that really a message that should be printed to the user at every save/file open?
2) the analysis indicator has no way to get this info without getting extra classes in the CTOR

Reply:
`The read takes the bindingConfiguration only to write logging`
What do you mean by this? We need this to both finding the mode and finding the file path. File path is calculated based on bindingconfiguration.